JASP-Engine/JASP/R/ttestindependentsamples.R

```diff
@@ -1,698 +1,695 @@
+            
+            variableData <- dataset[[ .v(variable) ]]
+
+            ## test is a number, indicating which tests should be run
+            for (test in seq_len(length(which_tests))) {
                 
-                    foot.index <- .addFootnote(footnotes, "Too few datapoints (N < 3) to compute statistic reliably")
-					row.footnotes <- list(W=list(foot.index), p=list(foot.index))
-					error <- TRUE
-					
-                } else if(length(data) > 5000) {
+                current_test <- which_tests[[test]]
                 
-                    foot.index <- .addFootnote(footnotes, "Too many datapoints (N > 5000) to compute statistic reliably")
-					row.footnotes <- list(W=list(foot.index), p=list(foot.index))
-					error <- TRUE
-					
+                ## don't run a test the user doesn't want
+                if (!current_test) {
+                    next
                 }
-                
-                if (!error) {
-                
-                    r <- stats::shapiro.test(data)
-                
-                    W <- .clean(as.numeric(r$statistic))
-                    p <- .clean(r$p.value)
-                
-                    if(level == levels[1]) {
-                        newGroup <- TRUE   
-                    } else {				
-                        newGroup <- FALSE
+
+                ## try to run the test, catching eventual errors
+                row <- try(silent = FALSE, expr = {
+
+                    row.footnotes <- NULL
+                    ci <- options$confidenceIntervalInterval # what a mouthful!
+                    f <- as.formula(paste(.v(variable), "~",
+                                          .v(options$groupingVariable)))
+
+                    if (test == 3) {
+                        what_test <- "Mann-Whitney"
+                        r <- stats::wilcox.test(f, data = dataset,
+                                                alternative = direction,
+                                                conf.int = TRUE, conf.level = ci, paired = FALSE)
+                        df <- ""
+                        m <- as.numeric(r$estimate)
+
+                    } else {
+                        what_test <- ifelse(test == 2, "Welch's", "Student's")
+                        r <- stats::t.test(f, data = dataset, alternative = direction,
+                                           var.equal = test != 2, conf.level = ci, paired = FALSE)
+
+                        df <- as.numeric(r$parameter)
+                        m <- as.numeric(r$estimate[1]) - as.numeric(r$estimate[2])
+                    }
+
+                    ## if the user doesn't want a Welch's t-test,
+                    ## give a footnote indicating if the equality of variance
+                    ## assumption is met; seems like in this setting there is no
+                    ## sampling plan, thus the p-value is not defined. haha!
+                    if (!wants_welchs && wants_students) {
```

Comment:
i would always place this warning beside the student's statistic if equality of variances is violated (perhaps not, if they're explicitly testing the assumption).